pyspark/bigdl/nn/keras/layer.py

```diff
@@ -58,6 +58,11 @@ def get_output_shape(self):
 
 
 class KerasLayer(Layer):
+    def __init__(self, jvalue, bigdl_type, name, *args):
+        super(Layer, self).__init__(jvalue, bigdl_type, *args)
+        if name:
+            self.set_name(name)
```

Comment:
we should propagate the name to scala side as well

Reply:
somehow we can inject the name on the init method of `Layer` instead of change the constructor in each subclass